src/datachain/lib/video.py

```diff
@@ -0,0 +1,273 @@
+import os.path
+import pathlib
+from typing import TYPE_CHECKING, Optional, Union
+
+from datachain.lib.file import VideoMeta
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+    from numpy import ndarray
+
+    from datachain.lib.file import VideoFile
+
+try:
+    import imageio.v3 as iio
+    from moviepy.video.io.VideoFileClip import VideoFileClip
+except ImportError as exc:
+    raise ImportError(
+        "Missing dependencies for processing video:\n"
+        "To install run:\n\n"
+        "  pip install 'datachain[video]'\n"
+    ) from exc
+
+
+def video_meta(file: "VideoFile") -> VideoMeta:
+    """
+    Returns video file meta information.
+
+    Args:
+        file (VideoFile): VideoFile object.
+
+    Returns:
+        VideoMeta: Video file meta information.
+    """
+    props = iio.improps(file.stream(), plugin="pyav")
+    frames_count, width, height, _ = props.shape
+
+    meta = iio.immeta(file.stream(), plugin="pyav")
```

Comment:
I don't like this part, it looks like we are reading video file twice here. Need to check the other way to get video meta information.

Reply:
yep, also are we reading the whole file to get meta?

Reply:
For now I have rewritten this code to use `ffmpeg-python` package, it is using file underneath. This is the fastest way I know to get video file metadata and more robust, since it is using `ffmpeg`. I am open to discussion here.